pkg/kf/commands/install/kf/kf.go

```diff
@@ -176,3 +194,82 @@ func installServiceCatalog(ctx context.Context) error {
 
 	return nil
 }
+
+// fetchReleases returns a list of releases YAML files from github and the
+// nightly. The nightly will be the first entry. The two slices are related.
+func fetchReleases(ctx context.Context) (names, addrs []string, err error) {
+	Logf(ctx, "fetching Kf releases...")
+	versions := []semver.Version{}
+	client := github.NewClient(nil)
+	releases, _, err := client.Repositories.ListReleases(ctx, "google", "kf", nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, r := range releases {
+		var name, tag, assetURL string
+		if r.Name != nil {
+			name = *r.Name
+		}
+
+		if r.TagName != nil {
+			tag = *r.TagName
+		}
+
+		for _, a := range r.Assets {
+			if a.Name != nil && a.BrowserDownloadURL != nil && *a.Name == "release.yaml" {
+				assetURL = *a.BrowserDownloadURL
+				break
+			}
+		}
+
+		if name == "" || assetURL == "" || tag == "" {
```

Comment:
If I'm reading right, these could be inlined to exit early up in the if statements on 211 and 215 which would make the control flow easier to understand e.g.

```
// Get the release name
if r.Name == nil {
	continue
}
name := *r.Name

// Parse the release tag for the version
if r.TagName == nil {
	continue // must have a tag
}
version, err = semvar.ParseTolerant(*r.TagName)
if err != nil {
    Logf(ctx, "invalid semver tag %q: %s", tag, err)
    continue
}

// Get the URL
// ...
```